dap/dap.hpp

```diff
@@ -566,9 +569,211 @@ struct WXDLLIMPEXP_DAP InitializeRequest : public Request {
     JSON_SERIALIZE();
 };
 
+#if __cplusplus >= 201703L  // AKA C++17
+/// A ColumnDescriptor specifies what module attribute to show in a column of the ModulesView,
+/// how to format it, and what the column’s label should be.
+/// It is only used if the underlying UI actually supports this level of customization.
+struct ColumnDescriptor {
+    // Name of the attribute rendered in this column.
+    wxString attributeName;
+
+    // Header UI label of column.
+    wxString label;
+
+    // Format to use for the rendered values in this column. TBD how the format
+    // strings looks like.
+    std::optional<wxString> format;
+
+    // Datatype of values in this column.  Defaults to 'string' if not specified.
+    //
+    // Must be one of the following enumeration values:
+    // 'string', 'number', 'boolean', 'unixTimestampUTC'
+    std::optional<wxString> type;
+
+    // Width of this column in characters (hint only).
+    std::optional<int> number;
+};
+
+/// An ExceptionBreakpointsFilter is shown in the UI as an filter option for
+/// configuring how exceptions are dealt with.
+struct ExceptionBreakpointsFilter {
+    // An optional help text providing information about the condition. This
+    // string is shown as the placeholder text for a text box and must be
+    // translated.
+    std::optional<wxString> conditionDescription;
+    // Initial value of the filter option. If not specified a value 'false' is
+    // assumed.
+    std::optional<bool> def;
+    // An optional help text providing additional information about the exception
+    // filter. This string is typically shown as a hover and must be translated.
+    std::optional<wxString> description;
+    // The internal ID of the filter option. This value is passed to the
+    // 'setExceptionBreakpoints' request.
+    wxString filter;
+    // The name of the filter option. This will be shown in the UI.
+    wxString label;
+    // Controls whether a condition can be specified for this filter option. If
+    // false or missing, a condition can not be set.
+    std::optional<bool> supportsCondition;
+};
+
+// Names of checksum algorithms that may be supported by a debug adapter.
+//
+// Must be one of the following enumeration values:
+// 'MD5', 'SHA1', 'SHA256', 'timestamp'
+using ChecksumAlgorithm = wxString;
+
+/// Information about the capabilities of a debug adapter.
+struct Capabilities {
```

Comment:
all structs should be decorated with `WXDLLIMPEXP_DAP `

Reply:
Just pushed the changes to fix this.